Libraries/Opc.Ua.Security.Certificates/Extensions/X509CrlNumberExtension.cs

```diff
@@ -0,0 +1,176 @@
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+
+namespace Opc.Ua.Security.Certificates
+{
+    /// <summary>
+    /// The CRL Number extension.
+    /// </summary>
+    /// <remarks>
+    ///    id-ce-cRLNumber OBJECT IDENTIFIER::= { id-ce 20 }
+    ///         CRLNumber::= INTEGER(0..MAX)
+    /// </remarks>
+    public class X509CrlNumberExtension : X509Extension
+    {
+        #region Constructors
+        /// <summary>
+        /// Creates an empty extension.
+        /// </summary>
+        protected X509CrlNumberExtension()
+        {
+        }
+
+        /// <summary>
+        /// Creates an extension from ASN.1 encoded data.
+        /// </summary>
+        public X509CrlNumberExtension(AsnEncodedData encodedExtension, bool critical)
+            : this(encodedExtension.Oid, encodedExtension.RawData, critical)
+        {
+        }
+
+        /// <summary>
+        /// Creates an extension from an Oid and ASN.1 encoded raw data.
+        /// </summary>
+        public X509CrlNumberExtension(string oid, byte[] rawData, bool critical)
+            : this(new Oid(oid, kFriendlyName), rawData, critical)
+        {
+        }
+
+        /// <summary>
+        /// Creates an extension from ASN.1 encoded data.
+        /// </summary>
+        public X509CrlNumberExtension(Oid oid, byte[] rawData, bool critical)
+        :
+            base(oid, rawData, critical)
+        {
+            Decode(rawData);
+        }
+
+        /// <summary>
+        /// Build the CRL Number extension (for CRL extensions).
+        /// </summary>
+        public X509CrlNumberExtension(BigInteger crlNumber)
+        {
+            Oid = new Oid(CrlNumberOid, kFriendlyName);
+            Critical = false;
+            CrlNumber = crlNumber;
+            RawData = Encode();
+        }
+        #endregion
+
+        #region Overridden Methods
+        /// <summary>
+        /// Returns a formatted version of the Abstract Syntax Notation One (ASN.1)-encoded data as a string.
+        /// </summary>
+        public override string Format(bool multiLine)
+        {
+            StringBuilder buffer = new StringBuilder();
+            buffer.Append(kFriendlyName);
+            buffer.Append("=");
+            buffer.Append(CrlNumber);
+
+            return buffer.ToString();
+        }
+
+        /// <summary>
+        /// Initializes the extension from ASN.1 encoded data.
+        /// </summary>
+        public override void CopyFrom(AsnEncodedData asnEncodedData)
+        {
+            if (asnEncodedData == null) throw new ArgumentNullException(nameof(asnEncodedData));
+            Oid = asnEncodedData.Oid;
+            RawData = asnEncodedData.RawData;
+            Decode(RawData);
+        }
+        #endregion
+
+        #region Public Properties
+        /// <summary>
+        /// The OID for a CRL Number extension.
+        /// </summary>
+        public const string CrlNumberOid = "2.5.29.20";
```

Comment:
Why not defining those constants in Oids class?

Reply:
same